### PR TITLE
Validate client options. Related to MONGOID-4157

### PR DESCRIPTION
**Pull request by estolfo**

### CODE REVIEW

**Comment by durran** [comment]
Just a thought - when invalid options are provided in a URI, we log it. I think for consistency it might be good to do it here as well.

**Comment by estolfo** [comment]
@durran done

**Comment by durran** [comment]
:+1:

File: lib/mongo/client.rb [hunk 1]
```diff
@@ -27,6 +28,40 @@ class Client
     # @since 2.1.0
     CRUD_OPTIONS = [ :database, :read, :write ].freeze
 
+    # Valid client options.
+    #
+    # @since 2.1.2
+    VALID_OPTIONS = [
+        :auth_mech,
```
Comment (durran) [inline lib/mongo/client.rb]: I'd do a 2 space indent here as it's standard formatting in Ruby, and then freeze the array.

Comment (estolfo) [inline lib/mongo/client.rb]: sorry, my ide puts in that spacing automatically. I'll update it.

**Pull request merged**